examples/preprocessing/plot_resample.py

```diff
@@ -0,0 +1,55 @@
+"""
+===============
+Resampling data
+===============
+
+Resampling can save memory and computation time. This example shows some
+approaches to resample data from 600 Hz to 100 Hz.
+"""
+# Authors: Marijn van Vliet <w.m.vanvliet@gmail.com>
+#
+# License: BSD (3-clause)
+#
+from __future__ import print_function
+
+import mne
+from mne.io import Raw
+from mne.datasets import sample
+
+print(__doc__)
+
+
+###############################################################################
+# The first approach is to resample the raw, continous data.
+
+data_path = sample.data_path()
+raw_fname = data_path + '/MEG/sample/sample_audvis_raw.fif'
+raw = Raw(raw_fname, preload=True)
+print('Original sampling rate:', raw.info['sfreq'], 'Hz')
+
+# Resample to 300 Hz 
+raw_resampled = raw.resample(300, copy=True)
+print('New sampling rate:', raw.info['sfreq'], 'Hz')
+
+###############################################################################
+# Because resampling also affects the stim channels, some trigger onsets might
+# be lost. While MNE attempts to downsample the stim channels in an intelligent
+# manner to avoid this, the recommended approach is to find events on the
+# original data before downsampling.
+print('Number of events before resampling:', len(mne.find_events(raw)))
+
+# Resample to 100 Hz (generates warning)
+raw_resampled = raw.resample(100, copy=True)
+print('Number of events after resampling:',
+      len(mne.find_events(raw_resampled)))
+
+# To avoid losing events, jointly resample the data and event matrix
+events = mne.find_events(raw)
+raw_resampled, events_resampled = raw.resample(100, events=events, copy=True)
+print('Number of events after resampling:', len(events_resampled))
+
+###############################################################################
+# Epoched data can be resampled as well.
```

Comment:
Think of a signal that you want to downsample by a factor of 10, but has an "event" at sample number 5 in the original data. Epoching first, your "sample zero" will correctly be at sample number 5, and then you will decimate/downsample that epoch, meaning the first sample of that epoch comes from what was originally sample number 5.

If you downsample then epoch, your "sample zero" at the new rate will have to come from sample number 0 or 10 at the original rate (because these are the only samples you have access to now), so you've introduced a jitter of 5 samples (at the original rate) or half a sample (at the new rate) that wouldn't be there in the other regime. Make sense?